stdlib/@tests/test_cases/check_dataclasses.py

```diff
@@ -41,6 +41,21 @@ class Foo:
     assert_type(f, Foo)
 
 
+def is_dataclass_any(arg: Any) -> None:
+    if dc.is_dataclass(arg):
+        assert_type(arg, Union["DataclassInstance", Type["DataclassInstance"]])
+
+
+def is_dataclass_object(arg: object) -> None:
+    if dc.is_dataclass(arg):
+        assert_type(arg, Union["DataclassInstance", Type["DataclassInstance"]])
+
+
+def is_dataclass_type(arg: type) -> None:
+    if dc.is_dataclass(arg):
+        assert_type(arg, Type["DataclassInstance"])
```

Comment:
Actually these cases are already checked by `check_other_isdataclass_overloads` below and can be removed.

```suggestion
```